tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ServicesCommand.java

```diff
@@ -0,0 +1,286 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.modules;
+
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+import aQute.remote.api.Agent;
+import aQute.remote.api.Event;
+import aQute.remote.api.Supervisor;
+import aQute.remote.util.AgentSupervisor;
+
+/**
+ * @author Lovett Li
+ */
+public class ServicesCommand {
+
+    private String _serviceName;
+
+    public ServicesCommand()
+    {
+    }
+    public ServicesCommand( String serviceName )
+    {
+        _serviceName = serviceName;
+    }
+
+    public String[] execute() throws Exception
+    {
+        ServicesSupervisor supervisor = null;
+        String[] result = null;
+        try
+        {
+            supervisor = connectRemote();
+
+            if( _serviceName == null )
+            {
+                result = getServices( supervisor );
+            }
+            else
+            {
+                result = getServiceBundle( _serviceName, supervisor );
+            }
+
+            return result;
+        }
+        finally
+        {
+            if( supervisor != null )
+            {
+                supervisor.close();
+            }
+
+        }
+    }
+
+    private ServicesSupervisor connectRemote() throws Exception
+    {
+        if( !canConnect( "localhost", Agent.DEFAULT_PORT ) )
+        {
+            return null;
+        }
+
+        ServicesSupervisor supervisor = new ServicesSupervisor();
+        supervisor.connect( "localhost", Agent.DEFAULT_PORT );
```

Comment:
Does this interfere with the running BundleSupervisor that is connected to the agent from the server launch?  In other words, can two different clients connect to the agent?